### PR TITLE
Added --version option (#1291)

### PR DESCRIPTION
**Pull request by andraskvr**

### CODE REVIEW

File: tests/test_app.py [hunk 1]
```diff
@@ -436,3 +439,18 @@ def test_warning_config_options_on_compile_command(snippetcompiler, tmpdir):
...
     assert f"Config directory {non_existing_config_dir} doesn't exist" not in all_output
     assert f"Config file {non_existing_config_file} doesn't exist" not in all_output
+
+
+@pytest.mark.parametrize(
+    "with_tty, regexes_required_lines, regexes_forbidden_lines",
+    [(False, [r"Current Inmanta version:"], []), (True, [r"Current Inmanta version:"], [])],
+)
+@pytest.mark.timeout(20)
+def test_version_argument_is_set(tmpdir, with_tty, regexes_required_lines, regexes_forbidden_lines):
```
Comment (arnaudsjs) [inline tests/test_app.py]: This test case only tests if the output starts with "Current Inmanta version:". We should verify the version number as well. Otherwise we are never sure if the "--version" command outputs the actual version number. 

Maybe it's also a good idea to test the negative scenario. I.e. test if version is not shown when the --version option is not present.

File: src/inmanta/app.py [hunk 2]
```diff
@@ -502,6 +506,11 @@ def cmd_parser():
     return parser
 
 
+def get_current_version_and_exit():
+    print("Current Inmanta version:", get_compiler_version())
```
Comment (arnaudsjs) [inline src/inmanta/app.py]: Minor: We try to use f-strings as much as possible is we need to parameterize a string. This improves readability and keeps code style of our project more consistent.

File: tests/test_app.py [hunk 3]
```diff
@@ -436,3 +439,18 @@ def test_warning_config_options_on_compile_command(snippetcompiler, tmpdir):
     assert "Compile done" in all_output
     assert f"Config directory {non_existing_config_dir} doesn't exist" not in all_output
     assert f"Config file {non_existing_config_file} doesn't exist" not in all_output
+
+
+@pytest.mark.parametrize(
+    "with_tty, regexes_required_lines, regexes_forbidden_lines",
+    [(False, [r"Current Inmanta version:"], []), (True, [r"Current Inmanta version:"], [])],
```
Comment (arnaudsjs) [inline tests/test_app.py]: The second and the third parameter are always the same, so it doesn't make sense to include it in the parameterization.

Comment (andraskvr) [reply tests/test_app.py]: Good point

File: tests/test_app.py (hunk 1 above)
Comment (wouterdb) [reply tests/test_app.py]: @arnaudsjs where is the testcase going to get the number?

Comment (arnaudsjs) [reply tests/test_app.py]: => get_compiler_version()

File: src/inmanta/app.py (hunk 2 above)
Comment (andraskvr) [reply src/inmanta/app.py]: Thanks for the notice, sure thing.
Is there a way to provide some warning about it and the other conventions as well (that are not covered by pep8)? Or it's not worth the effort (right now)?

File: tests/test_app.py (hunk 1 above)
Comment (andraskvr) [reply tests/test_app.py]: But the source implementation does the same, they could both be wrong and the test would pass

File: tests/test_app.py [hunk 4]
```diff
@@ -436,3 +439,23 @@ def test_warning_config_options_on_compile_command(snippetcompiler, tmpdir):
...
     assert f"Config directory {non_existing_config_dir} doesn't exist" not in all_output
     assert f"Config file {non_existing_config_file} doesn't exist" not in all_output
+
+
+@pytest.mark.parametrize(
+    "with_tty, version_should_be_shown, regexes_required_lines, regexes_forbidden_lines",
+    [
+        (False, True, [r"Current Inmanta version:"], []),
+        (True, True, [r"Current Inmanta version:"], []),
+        (True, False, [], [r"Current Inmanta version:"]),
```
Comment (arnaudsjs) [inline tests/test_app.py]:
```suggestion
        (False, False, [], [r"Current Inmanta version:"]),
```
